src/client-configuration.ts

```diff
@@ -204,6 +204,63 @@ export type StreamClientConfiguration = {
   cursor?: string;
 };
 
+/**
+ * Configuration for a change feed client. This typically comes from the `Client`
+ * instance configuration.
+ */
+export type ChangeFeedClientConfiguration = {
+  /**
+   * The underlying {@link HTTPClient} that will execute the actual HTTP calls
+   */
+  httpClient: HTTPClient;
+
+  /**
+   * Controls what Javascript type to deserialize {@link https://docs.fauna.com/fauna/current/reference/fql_reference/types#long | Fauna longs} to.
+   *
+   * @see {@link ClientConfiguration.long_type}
+   */
+  long_type: "number" | "bigint";
```

Comment:
In beta, before we had the `long_type` config, we always deserialized Fauna `Long` types to BigInts, but we got feedback from folks that they want more control. We added a warning at the same time we added the new config.

https://github.com/fauna/fauna-js/blob/80b897eba674b965d5efc64a2897cf9c0db265f9/src/tagged-type.ts#L111-L120

Are you proposing that we effectively change that console.warn into thrown error? I think worth considering if we are going to make a major version bump at some point. But I also think it's outside the scope of this PR. I wouldn't want to make change/event feeds work differently than queries and streams.

Reply:
i am proposing either:
- add a flag to upgrade from warn to throw (non-breaking) for all deserialization (change feeds, queries, and streams)
- changing the default behavior from warn to throw (breaking) for all deserialization (change feeds, queries, and streams)

this is likely going to be an insidious bug in business logic for customers at best, but because it affects non-stringified IDs, it can also bite in really simple, business-logic-agnostic use-cases like "a UDF that returns an ID, and a client that updates the thing at that ID."

Reply:
Let's solve that in a separate PR. I'll do the follow-up, and we can start with the backwards compat opt-in until the next major version.